db/db_impl/db_impl_open.cc

```diff
@@ -194,6 +194,16 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src,
   }
 #endif  // !ROCKSDB_LITE
 
+  // Supported wal compression types
```

Comment:
We can just check ```result.wal_compression``` rather than going the roundabout way.

Reply:
Addressed.